src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1408,99 +1409,177 @@ static constexpr bool _isHexNumber(const wchar_t wch) noexcept
 
 // Routine Description:
 // - Given a color spec string, attempts to parse the color that's encoded.
-//   The only supported spec currently is the following:
-//      spec: a color in the following format:
+//   The supported specs currently are the following:
+//      spec1: a color in the following format:
 //          "rgb:<red>/<green>/<blue>"
-//          where <color> is one or two hex digits, upper or lower case.
+//      spec2: a color in the following format:
+//          "#<red><green><blue>"
+//
+//   In both specs, <color> is a value contains up to 4 hex digits, upper or lower case.
 // Arguments:
 // - string - The string containing the color spec string to parse.
 // - rgb - receives the color that we parsed
 // Return Value:
 // - True if a color was successfully parsed
 bool OutputStateMachineEngine::s_ParseColorSpec(const std::wstring_view string,

```

Comment:
Here's a crazy idea.

I think we should move this into utils.cpp right under `ColorFromHexString`.

This will move us one step closer to having a unified color parser, just like `XParseColor`.

We can then start to make use of it in other parts of the code, like Terminal's settings handler, to get support for "any kind of color the terminal can handle".

Reply:
We could then put specific tests into the TypesTests/UtilsTests.

Reply:
It could return an `std::optional<til::color>`.

Reply:
it may require a copy of `s_HexToUint`.

Reply:
That’s actually not a bad idea. I was feeling that it’s making output engine bloated and wanted to move it. But I think the ParseColor should still be separated from ColorFromHexString. If there’s need for a universal color parser we should make a new one by combining those two.

Reply:
Your code below shows that they can be easily unified! Maybe it's better to have them unified instead of having two separate ones? Otherwise, we will need to make sure `ColorFromHexString` supports `#rrrgggbbb` and `#rrrrggggbbbb` at the same time... hmm

Reply:
Oh you said “right under” pardon my understanding.

Reply:
Well you do want to unify them? I don’t know man. The “rgb:R/G/B” isn’t really hex string right? Or is it?...

Reply:
We can put them in the same place today, and we can consider unifying them later! No need to unify them right now. :smile: